### ISSUE TITLE
Spot-count histogram truncates images

### ISSUE THREAD

**Issue by rjgildea** [dials/dials#1659]
E.g. for a 3-image sweep, spotfinding is performed on all 3 images, but the ascii spot-count histogram only reports the first 2 images:
```
$ dials.import $(dials.data get -q x4wide)/X4_wide_M1S4_2_*.cbf image_range=1,3
$ dials.find_spots imported.expt
...
Finding spots in image 1 to 3...
...
Found 12622 strong pixels on image 1
Found 12652 strong pixels on image 2
Found 11451 strong pixels on image 3
...
Histogram of per-image spot count for imageset 0:
958 spots found on 2 images (max 514 / bin)
* 
**
**
**
**
**
**
**
**
**
12
```

**Comment by graeme-winter** [comment]
Yeh, this is well known, and is part of the whole general mish mask of off by one errors.

I have tried to fix this on several occasions. I failed.

**Comment by graeme-winter** [comment]
When you peel this onion you find at it's core a lot of nasty `dxtbx` things - all those assumptions about images counting from 1 not 0, etc. 

I'll be delighted if you want to fix this, but I think it will be found to be a problem five times larger or more than you would expect.... unless I missed something obvious, repeatedly.

**Comment by rjgildea** [comment]
Fixed by a66ee17 (#1660)
